### PR TITLE
[Mellanox] Add support for SN4280 platform

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

Add support for SN4280 platform

| Port configuration | Value |
| ------  |--------- |
 | Breakout mode for each port  |**Defined in port mapping** |
| Speed of the port |  **Defined in Port mapping** |
| Internal Ports |  **Defined in Port mapping**|

 Buffer configuration | Value
------  |---------
 Shared headroom | **Enabled**
 Shared headroom pool factor | **2**
 Dynamic Buffer | **Disable**
 In static buffer scenario how many uplinks and downlinks? | **8 1x400G  uplinks and 20 1x400G downlinks, 4 1x400G downlink Internal ports**

Port Mapping

| Ports  | Mode      |
| ------  |--------- |
| 1-8   | 1x400G  |
| 9-28  | 1x400G   | 
| 28-32  | 1x400G (Internal Ports connected to DPU) |

Number of Uplinks / Downlinks:

Default topology: T1

T1-T2: 2000m
T0-T1: 300m
T0-server: 40m

#### How to verify it

Verified by running regression on SN4280 platform

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/sai.profile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/sai_4280.xml
+SAI_DEFAULT_SWITCHING_MODE_STORE_FORWARD=1
```
Comment (dgsudharsan) [inline device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/sai.profile:2]: Please define this as well as SAI_NOT_DROP_SIP_DIP_LINK_LOCAL=1 only in Mellanox-SN4280-O28 SKU

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+This SKU is dedicated to Smart Switch
```
Comment (dgsudharsan) [inline device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/README.md:1]: Why is this file needed? I don't see it in other SKUs

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/sai_4280.xml [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+                <!-- (BITMASK) 2 - 1Gb , 16 - 10Gb , 32 - 40Gb , 384 - 50Gb , 1536 - 100Gb , 4096 - 200Gb -->
+                <port-speed>4096</port-speed>
+            </port-info>
+            <port-info>
+                <local-port>5</local-port>
+                <width>8</width>
+                <module>16</module>
+                <breakout-modes>3</breakout-modes>
+                <port-speed>4096</port-speed>
```
Comment (dgsudharsan) [inline device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/sai_4280.xml:51]: Should this be 32768?

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/sai_4280.xml [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+                <module>17</module>
+
+                <!-- 0 none, 1=2, 2=4, 3=2,4 -->
+                <breakout-modes>3</breakout-modes>
+
+                <!-- (BITMASK) 2 - 1Gb , 16 - 10Gb , 32 - 40Gb , 384 - 50Gb , 1536 - 100Gb , 4096 - 200Gb -->
+                <port-speed>4096</port-speed>
+            </port-info>
+            <port-info>
+                <local-port>5</local-port>
```
Comment (dgsudharsan) [inline device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/sai_4280.xml:47]: Can you check if the local port needs to have a gap of 8 instead of 4.

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/README.md (hunk 2 above)
Comment (vivekrnv) [reply device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/README.md]: removed

File: device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/sai_4280.xml (hunk 3 above)
Comment (vivekrnv) [reply device/mellanox/x86_64-nvidia_sn4280-r0/ACS-SN4280/sai_4280.xml]: updated

**Review by dgsudharsan — APPROVED** [review]

**Comment by vivekrnv** [comment]
@stephenxs Please review buffer changes

**Comment by stephenxs** [comment]
> @stephenxs Please review buffer changes

Hi @vivekrnv 
we probably need to update headroom (pg_profile_lookup.ini) file starting from this SKU.
I'm asking @baiwei0427 to provide the proper percentage to use.
Will get back to you once I have an answer

**Comment by stephenxs** [comment]
> > @stephenxs Please review buffer changes
> 
> Hi @vivekrnv we probably need to update headroom (pg_profile_lookup.ini) file starting from this SKU. I'm asking @baiwei0427 to provide the proper percentage to use. Will get back to you once I have an answer

50%.

**Comment by vivekrnv** [comment]
> > @stephenxs Please review buffer changes
> 
> Hi @vivekrnv we probably need to update headroom (pg_profile_lookup.ini) file starting from this SKU. I'm asking @baiwei0427 to provide the proper percentage to use. Will get back to you once I have an answer

Hi, Updated the values. please check

**Comment by stephenxs** [comment]
buffer pool sizes LGTM. How does it guarantee the cable length is deployed the expected cable length? currently, the default value differs. see https://github.com/sonic-net/sonic-buildimage/blob/master/files/build_templates/buffers_config.j2#L41

**Review by stephenxs — APPROVED** [review]

**Review by stephenxs — APPROVED** [review]
